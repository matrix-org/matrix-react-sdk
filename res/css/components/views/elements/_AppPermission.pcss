/*
Copyright 2015, 2016 OpenMarket Ltd
Copyright 2019 - 2023 The Matrix.org Foundation C.I.C.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.mx_AppPermission {
    > div {
        margin-bottom: 12px;
    }

    h4 {
        margin: 0;
        padding: 0;
    }

    .mx_AppPermission_smallText {
        font-size: $font-12px;
    }

    .mx_AppPermission_bolder {
        font-weight: var(--font-semi-bold);
    }

    .mx_AppPermission_helpIcon {
        margin-top: 1px;
        margin-right: 2px;
        width: 10px;
        height: 10px;
        display: inline-block;

        &::before {
            display: inline-block;
            background-color: $accent;
            mask-repeat: no-repeat;
            mask-size: 12px;
            width: 12px;
            height: 12px;
            mask-position: center;
            content: "";
            vertical-align: middle;
            mask-image: url("$(res)/img/feather-customised/help-circle.svg");
        }
    }
}

.mx_Tooltip.mx_Tooltip--appPermission {
    box-shadow: none;
    background-color: $tooltip-timeline-bg-color;
    color: $tooltip-timeline-fg-color;
    border: none;
    border-radius: 3px;
    padding: 6px 8px;

    &.mx_Tooltip--appPermission--dark {
        .mx_Tooltip_chevron::after {
            border-right-color: $tooltip-timeline-bg-color;
        }
    }

    ul {
        list-style-position: inside;
        padding-left: 2px;
        margin-left: 0;
    }
}
