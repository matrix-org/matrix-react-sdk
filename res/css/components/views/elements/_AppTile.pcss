/*
Copyright 2023 Suguru Hirahara

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

:root {
    --AppTile--mini-height: 220px;
}

.mx_AppTile {
    display: flex;
    flex-direction: column;

    .mx_AppTile_menuBar {
        margin: 0;
        font-size: $font-12px;
        background-color: $widget-menu-bar-bg-color;
        display: flex;
        align-items: center;
        justify-content: space-between;
        width: 100%;
        padding-top: 3px;
        padding-bottom: 6px;

        .mx_AppTile_menuBar_title {
            line-height: 20px;
            white-space: nowrap;
            overflow: hidden;
            text-overflow: ellipsis;

            .mx_WidgetAvatar {
                margin-right: 12px;
            }

            > :last-child {
                margin-left: 9px;
            }
        }

        .mx_AppTile_menuBar_widgets {
            display: flex;
            align-items: center;

            .mx_AppTile_menuBar_widgets_button {
                --size: 24px; /* Size of the button. Its height and width values should be same */

                margin: 0 4px;
                position: relative;
                height: var(--size);
                width: var(--size);
                display: flex;
                align-items: center;
                justify-content: center;

                &:hover::after {
                    content: "";
                    position: absolute;
                    height: var(--size);
                    width: var(--size);
                    background-color: $panel-actions;
                    border-radius: 50%;
                    left: 0;
                    top: 0;
                }

                .mx_Icon {
                    color: $muted-fg-color;
                }
            }
        }
    }

    .mx_AppTile_persistedWrapper {
        flex: 1;

        div {
            width: 100%;
            height: 100%;
        }
    }

    /* appTileBody is embedded to PersistedElement outside of mx_AppTile,
       so rules to style appTileBody generally should not be included here. */
    .mx_AppTileBody--large,
    .mx_AppTileBody--mini {
        height: inherit;
        flex: 1;
    }

    &.mx_AppTile--large,
    &.mx_AppTile--fullWidth {
        border: var(--container-border-width) solid $widget-menu-bar-bg-color;
        background-color: $widget-menu-bar-bg-color;
    }

    &.mx_AppTile--large {
        box-sizing: border-box;
    }

    &.mx_AppTile--fullWidth {
        width: 100% !important; /* to override the inline style set by the resizer */
        border-radius: 8px;
    }

    &.mx_AppTile--mini {
        width: 100%;
        height: var(--AppTile--mini-height);
    }
}

/* Rules added to this selector style appTileBody generally */
.mx_AppTileBody {
    /* Apply to every variant of appTileBody */
    border-radius: 8px;

    /* const loadingElement */
    /* Note the loading spinner and the message next to it are not always included in mx_AppTileBody--loading */
    .mx_AppTileBody_fadeInSpinner {
        /* place spinner and the message at the center of mx_AppTileBody */
        height: 100%;
        width: 100%;

        font-weight: bold; /* message next to the spinner */
        animation-fill-mode: backwards;
        animation-duration: 200ms;
        animation-delay: 500ms;
        animation-name: mx_AppTileBody_fadeInSpinnerAnimation;
    }

    &.mx_AppTileBody--large,
    &.mx_AppTileBody--mini {
        width: 100%;
        overflow: hidden;
        height: var(--AppTileBody-height);

        iframe {
            border: none;
            width: 100%;
            height: 100%;
        }
    }

    &.mx_AppTileBody--large {
        --AppTileBody-height: 100%;

        background-color: $widget-body-bg-color;

        iframe {
            overflow: hidden;
            padding: 0;
            margin: 0;
            display: block;
        }
    }

    &.mx_AppTileBody--mini {
        --AppTileBody-height: var(--AppTile--mini-height);
    }

    &.mx_AppTileBody--loading {
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        position: relative;
        height: 100%;

        /* match bg of border so that the cut corners have the right fill */
        background-color: $widget-body-bg-color !important;

        iframe {
            display: none;
        }
    }
}

@keyframes mx_AppTileBody_fadeInSpinnerAnimation {
    from {
        opacity: 0;
    }
    to {
        opacity: 1;
    }
}
