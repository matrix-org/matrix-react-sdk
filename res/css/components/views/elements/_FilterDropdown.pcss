/*
Copyright 2022 The Matrix.org Foundation C.I.C.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.mx_FilterDropdown {
    .mx_Dropdown_menu {
        margin-top: $spacing-4;
        left: unset;
        right: -$spacing-12;
        width: 232px;
        padding: $spacing-12;

        border: 1px solid $quinary-content;
        border-radius: 8px;
        box-shadow: 0px 1px 3px rgba(23, 25, 28, 0.05);

        background-color: $system;

        .mx_Dropdown_option_highlight {
            background-color: transparent;
        }
    }

    .mx_Dropdown_input {
        height: 24px;
        background-color: transparent;
        border-color: transparent;
        color: $secondary-content;
        border-radius: 4px;

        &:focus,
        &:hover {
            background-color: $quinary-content;
            border-color: $quinary-content;
        }
    }

    .mx_Dropdown_arrow {
        background: $secondary-content;
    }
}

.mx_FilterDropdown_option {
    position: relative;
    width: 100%;
    box-sizing: border-box;
    padding: $spacing-8 0 $spacing-8 $spacing-20;

    font-size: $font-12px;
    line-height: $font-15px;
    color: $primary-content;
}

.mx_FilterDropdown_optionSelectedIcon {
    height: 14px;
    width: 14px;
    position: absolute;
    top: $spacing-8;
    left: 0;
}

.mx_FilterDropdown_optionLabel {
    font-weight: var(--cpd-font-weight-semibold);
    display: block;
}

.mx_FilterDropdown_optionDescription {
    color: $secondary-content;
    margin-top: $spacing-4;
}
