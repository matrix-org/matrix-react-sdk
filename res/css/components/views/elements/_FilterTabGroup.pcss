/*
Copyright 2023 The Matrix.org Foundation C.I.C.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.mx_FilterTabGroup {
    color: $primary-content;
    label {
        margin-right: $spacing-12;
        cursor: pointer;
        span {
            display: inline-block;
            line-height: $font-24px;
        }
    }
    input[type="radio"] {
        appearance: none;
        margin: 0;
        padding: 0;

        &:focus,
        &:hover {
            & + span {
                color: $secondary-content;
            }
        }

        &:checked + span {
            color: $accent;
            font-weight: var(--cpd-font-weight-semibold);
            // underline
            box-shadow: 0 1.5px 0 0 currentColor;
        }
    }
}
