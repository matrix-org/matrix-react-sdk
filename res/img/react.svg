<svg xmlns="http://www.w3.org/2000/svg" width="18" height="17" viewBox="0 0 18 17">
    <g fill="none" fill-rule="evenodd">
        <path fill="#2E2F32" fill-rule="nonzero" d="M13.6 14.5a7.875 7.875 0 1 1 2.183-3h-.818a7.125 7.125 0 1 0-2.62 3H13.6z"/>
        <g stroke="#2E2F32" stroke-linecap="round" stroke-linejoin="round" stroke-width=".75">
            <path d="M11.5 13h6M14.5 10v6"/>
        </g>
        <path stroke="#2E2F32" stroke-linecap="round" stroke-linejoin="round" stroke-width=".75" d="M5.5 10s1.125 1.5 3 1.5 3-1.5 3-1.5"/>
        <path stroke="#2E2F32" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" d="M6.25 6.25h.008M10.75 6.25h.008"/>
    </g>
</svg>
