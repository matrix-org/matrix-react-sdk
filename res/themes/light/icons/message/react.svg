<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><path d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 464c-119.1 0-216-96.9-216-216S128.9 40 248 40s216 96.9 216 216-96.9 216-216 216zm90.2-146.2C315.8 352.6 282.9 368 248 368s-67.8-15.4-90.2-42.2c-5.7-6.8-15.8-7.7-22.5-2-6.8 5.7-7.7 15.7-2 22.5C161.7 380.4 203.6 400 248 400s86.3-19.6 114.8-53.8c5.7-6.8 4.8-16.9-2-22.5-6.8-5.6-16.9-4.7-22.6 2.1zM168 240c17.7 0 32-14.3 32-32s-14.3-32-32-32-32 14.3-32 32 14.3 32 32 32zm160 0c17.7 0 32-14.3 32-32s-14.3-32-32-32-32 14.3-32 32 14.3 32 32 32z"/></svg>