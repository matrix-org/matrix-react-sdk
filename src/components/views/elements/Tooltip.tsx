/*
Copyright 2015, 2016 OpenMarket Ltd
Copyright 2019 New Vector Ltd
Copyright 2019 Michael Telatynski <7t3chguy@gmail.com>
Copyright 2019, 2020 The Matrix.org Foundation C.I.C.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/


import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import dis from '../../../dispatcher/dispatcher';
import classNames from 'classnames';
import { ViewTooltipPayload } from '../../../dispatcher/payloads/ViewTooltipPayload';
import { Action } from '../../../dispatcher/actions';

const MIN_TOOLTIP_HEIGHT = 25;

interface IProps {
        // Class applied to the element used to position the tooltip
        className: string,
        // Class applied to the tooltip itself
        tooltipClassName?: string,
        // Whether the tooltip is visible or hidden.
        // The hidden state allows animating the tooltip away via CSS.
        // Defaults to visible if unset.
        visible?: boolean,
        // the react element to put into the tooltip
        label: React.ReactNode,
}

export default class Tooltip extends React.Component<IProps> {
    private tooltipContainer: HTMLElement;
    private tooltip: void | Element | Component<Element, any, any>;
    private parent: Element;


    public static readonly defaultProps = {
        visible: true,
    };

    // Create a wrapper for the tooltip outside the parent and attach it to the body element
    public componentDidMount() {
        this.tooltipContainer = document.createElement("div");
        this.tooltipContainer.className = "mx_Tooltip_wrapper";
        document.body.appendChild(this.tooltipContainer);
        window.addEventListener('scroll', this.renderTooltip, true);

        this.parent = ReactDOM.findDOMNode(this).parentNode as Element;

        this.renderTooltip();
    }

    public componentDidUpdate() {
        this.renderTooltip();
    }

    // Remove the wrapper element, as the tooltip has finished using it
    public componentWillUnmount() {
        dis.dispatch<ViewTooltipPayload>({
            action: Action.ViewTooltip,
            tooltip: null,
            parent: null,
        });

        ReactDOM.unmountComponentAtNode(this.tooltipContainer);
        document.body.removeChild(this.tooltipContainer);
        window.removeEventListener('scroll', this.renderTooltip, true);
    }

    private updatePosition(style: {[key: string]: any}) {
        const parentBox = this.parent.getBoundingClientRect();
        let offset = 0;
        if (parentBox.height > MIN_TOOLTIP_HEIGHT) {
            offset = Math.floor((parentBox.height - MIN_TOOLTIP_HEIGHT) / 2);
        } else {
            // The tooltip is larger than the parent height: figure out what offset
            // we need so that we're still centered.
            offset = Math.floor(parentBox.height - MIN_TOOLTIP_HEIGHT);
        }
        style.top = (parentBox.top - 2) + window.pageYOffset + offset;
        style.left = 6 + parentBox.right + window.pageXOffset;
        return style;
    }

    private renderTooltip = () => {
        // Add the parent's position to the tooltips, so it's correctly
        // positioned, also taking into account any window zoom
        // NOTE: The additional 6 pixels for the left position, is to take account of the
        // tooltips chevron
        const parent = ReactDOM.findDOMNode(this).parentNode as Element;
        const style = this.updatePosition({});
        // Hide the entire container when not visible. This prevents flashing of the tooltip
        // if it is not meant to be visible on first mount.
        style.display = this.props.visible ? "block" : "none";

        const tooltipClasses = classNames("mx_Tooltip", this.props.tooltipClassName, {
            "mx_Tooltip_visible": this.props.visible,
            "mx_Tooltip_invisible": !this.props.visible,
        });

        const tooltip = (
            <div className={tooltipClasses} style={style}>
                <div className="mx_Tooltip_chevron" />
                { this.props.label }
            </div>
        );

        // Render the tooltip manually, as we wish it not to be rendered within the parent
        this.tooltip = ReactDOM.render<Element>(tooltip, this.tooltipContainer);

        // Tell the roomlist about us so it can manipulate us if it wishes
        dis.dispatch<ViewTooltipPayload>({
            action: Action.ViewTooltip,
            tooltip: this.tooltip,
            parent: parent,
        });
    }

    public render() {
        // Render a placeholder
        return (
            <div className={this.props.className} >
            </div>
        );
    }
}
