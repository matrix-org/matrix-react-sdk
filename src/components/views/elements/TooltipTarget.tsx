/*
Copyright 2021 The Matrix.org Foundation C.I.C.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React, { forwardRef, HTMLAttributes, useRef } from "react";
import { randomString } from "matrix-js-sdk/src/randomstring";

import useFocus from "../../../hooks/useFocus";
import useHover from "../../../hooks/useHover";
import Tooltip, { ITooltipProps } from "./Tooltip";

interface IProps
    extends HTMLAttributes<HTMLSpanElement>,
        Omit<ITooltipProps, "visible" | "tabIndex" | "aria-describedby"> {
    tooltipTargetClassName?: string;
    ignoreHover?: (ev: React.MouseEvent) => boolean;
}

/**
 * Generic tooltip target element that handles tooltip visibility state
 * and displays children
 */
const TooltipTarget = forwardRef<HTMLDivElement, IProps>(
    (
        {
            children,
            tooltipTargetClassName,
            // tooltip pass through props
            className,
            id,
            label,
            alignment,
            tooltipClassName,
            maxParentWidth,
            ignoreHover,
            ...rest
        },
        ref,
    ) => {
        const idRef = useRef("mx_TooltipTarget_" + randomString(8));
        // Use generated ID if one is not passed
        if (id === undefined) {
            id = idRef.current;
        }

        const [isFocused, focusProps] = useFocus();
        const [isHovering, hoverProps] = useHover(ignoreHover || (() => false));

        // No need to fill up the DOM with hidden tooltip elements. Only add the
        // tooltip when we're hovering over the item (performance)
        const tooltip = (isFocused || isHovering) && (
            <Tooltip
                id={id}
                className={className}
                tooltipClassName={tooltipClassName}
                label={label}
                alignment={alignment}
                visible={isFocused || isHovering}
                maxParentWidth={maxParentWidth}
            />
        );

        return (
            <div
                {...hoverProps}
                {...focusProps}
                tabIndex={0}
                aria-describedby={id}
                className={tooltipTargetClassName}
                {...rest}
                ref={ref}
            >
                {children}
                {tooltip}
            </div>
        );
    },
);

export default TooltipTarget;
