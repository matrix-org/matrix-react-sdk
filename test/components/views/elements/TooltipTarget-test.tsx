/*
Copyright 2022 The Matrix.org Foundation C.I.C.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from "react";
import { fireEvent, render } from "@testing-library/react";

import { Alignment } from "../../../../src/components/views/elements/Tooltip";
import TooltipTarget from "../../../../src/components/views/elements/TooltipTarget";

describe("<TooltipTarget />", () => {
    const defaultProps = {
        "tooltipTargetClassName": "test tooltipTargetClassName",
        "className": "test className",
        "tooltipClassName": "test tooltipClassName",
        "label": "test label",
        "alignment": Alignment.Left,
        "id": "test id",
        "data-testid": "test",
    };

    const getComponent = (props = {}) => {
        const wrapper = render(
            // wrap in element so renderIntoDocument can render functional component
            <span>
                <TooltipTarget {...defaultProps} {...props}>
                    <span>child</span>
                </TooltipTarget>
            </span>,
        );
        return wrapper.getByTestId("test");
    };

    const getVisibleTooltip = () => document.querySelector(".mx_Tooltip.mx_Tooltip_visible");

    it("renders container", () => {
        const component = getComponent();
        expect(component).toMatchSnapshot();
        expect(getVisibleTooltip()).toBeFalsy();
    });

    const alignmentKeys = Object.keys(Alignment).filter((o: any) => isNaN(o));
    it.each(alignmentKeys)("displays %s aligned tooltip on mouseover", async (alignment: any) => {
        const wrapper = getComponent({ alignment: Alignment[alignment] })!;
        fireEvent.mouseOver(wrapper);
        expect(getVisibleTooltip()).toMatchSnapshot();
    });

    it("hides tooltip on mouseleave", () => {
        const wrapper = getComponent()!;
        fireEvent.mouseOver(wrapper);
        expect(getVisibleTooltip()).toBeTruthy();
        fireEvent.mouseLeave(wrapper);
        expect(getVisibleTooltip()).toBeFalsy();
    });

    it("displays tooltip on focus", () => {
        const wrapper = getComponent()!;
        fireEvent.focus(wrapper);
        expect(getVisibleTooltip()).toBeTruthy();
    });

    it("hides tooltip on blur", async () => {
        const wrapper = getComponent()!;
        fireEvent.focus(wrapper);
        expect(getVisibleTooltip()).toBeTruthy();
        fireEvent.blur(wrapper);
        expect(getVisibleTooltip()).toBeFalsy();
    });
});
