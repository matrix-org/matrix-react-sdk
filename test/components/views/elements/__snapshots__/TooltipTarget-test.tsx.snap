// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TooltipTarget /> displays Bottom aligned tooltip on mouseover 1`] = `
<div
  class="mx_Tooltip test tooltipClassName mx_Tooltip_visible"
  id="test id"
  role="tooltip"
  style="display: block; top: 6px; transform: translate(max(10px, min(calc(0px - 50%), calc(100vw - 100% - 10px))));"
>
  <div
    class="mx_Tooltip_chevron"
  />
  test label
</div>
`;

exports[`<TooltipTarget /> displays InnerBottom aligned tooltip on mouseover 1`] = `
<div
  class="mx_Tooltip test tooltipClassName mx_Tooltip_visible"
  id="test id"
  role="tooltip"
  style="display: block; top: -50px; transform: translate(max(10px, min(calc(0px - 50%), calc(100vw - 100% - 10px))));"
>
  <div
    class="mx_Tooltip_chevron"
  />
  test label
</div>
`;

exports[`<TooltipTarget /> displays Left aligned tooltip on mouseover 1`] = `
<div
  class="mx_Tooltip test tooltipClassName mx_Tooltip_visible"
  id="test id"
  role="tooltip"
  style="display: block; right: 1030px; top: 0px; transform: translateY(-50%);"
>
  <div
    class="mx_Tooltip_chevron"
  />
  test label
</div>
`;

exports[`<TooltipTarget /> displays Natural aligned tooltip on mouseover 1`] = `
<div
  class="mx_Tooltip test tooltipClassName mx_Tooltip_visible"
  id="test id"
  role="tooltip"
  style="display: block; left: 6px; top: 0px; transform: translateY(-50%);"
>
  <div
    class="mx_Tooltip_chevron"
  />
  test label
</div>
`;

exports[`<TooltipTarget /> displays Right aligned tooltip on mouseover 1`] = `
<div
  class="mx_Tooltip test tooltipClassName mx_Tooltip_visible"
  id="test id"
  role="tooltip"
  style="display: block; left: 6px; top: 0px; transform: translateY(-50%);"
>
  <div
    class="mx_Tooltip_chevron"
  />
  test label
</div>
`;

exports[`<TooltipTarget /> displays Top aligned tooltip on mouseover 1`] = `
<div
  class="mx_Tooltip test tooltipClassName mx_Tooltip_visible"
  id="test id"
  role="tooltip"
  style="display: block; top: -6px; transform: translate(max(10px, min(calc(0px - 50%), calc(100vw - 100% - 10px))), -100%);"
>
  <div
    class="mx_Tooltip_chevron"
  />
  test label
</div>
`;

exports[`<TooltipTarget /> displays TopRight aligned tooltip on mouseover 1`] = `
<div
  class="mx_Tooltip test tooltipClassName mx_Tooltip_visible"
  id="test id"
  role="tooltip"
  style="display: block; top: -6px; right: 1024px; transform: translateY(-100%);"
>
  <div
    class="mx_Tooltip_chevron"
  />
  test label
</div>
`;

exports[`<TooltipTarget /> renders container 1`] = `
<div
  aria-describedby="test id"
  class="test tooltipTargetClassName"
  data-testid="test"
  tabindex="0"
>
  <span>
    child
  </span>
</div>
`;
